### PR TITLE
Auto day width, custom height

### PR DESCRIPTION
**Pull request by janbina**
Hi, firstly, thanks for this awesome library!
I found myself in a situation that I wanted day cell to have an automatic width to span the whole calendar width, but custom height at the same time. That was not possible with the library and even though it is possible to achieve that from the application, it might be quite difficult, whereas it is easy to do so in the library. So I created this fix for myself and I think others might find it useful as well.

### CODE REVIEW

**Comment by pratikKataria** [comment]
@janbina hey please help how can I set the height of day in activity

**Comment by janbina** [comment]
@pratikKataria Hey, is this related to this PR?
You can set the height of day from activity in `onCreate` like this:
`calendarView.daySize = Size(yourWidth, yourHeight)`

If you wanna use my modification, you can do it like this:
`calendarView.daySize = CalendarView.sizeAutoWidth(yourHeight)`
you will also need to use my fork, for example by using [jitpack](https://jitpack.io/#janbina/CalendarView/6b16bdb842)

**Comment by pratikKataria** [comment]
@janbina Thanks for the solution, but I come up with my own solution

File: library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt [hunk 1]
```diff
@@ -956,5 +962,11 @@ open class CalendarView : RecyclerView {
          * be the width of the calender divided by 7.
          */
         val SIZE_SQUARE = Size(DAY_SIZE_SQUARE, DAY_SIZE_SQUARE)
+
+        /**
+         * A value for [daySize] which indicates that the day cells should
+         * have width of the calender divided by 7 and provided height.
+         */
+        fun sizeAutoWidth(@Px height: Int) = Size(DAY_SIZE_SQUARE, height)
```
Comment (kizitonwose) [inline library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt:970]: Been thinking of a less-confusing name for this method for a while now. `sizeAutoWidth()` seems to have a lot of meaning. But looks like it's the only option in the end.

**Comment by kizitonwose** [comment]
Looks good, I'm open to more suggestions for the `sizeAutoWidth` method. In the absence of suggestions, I'll merge it this way.

**Comment by janbina** [comment]
I've been thinking about that as well. Another thing is that the name `DAY_SIZE_SQUARE` won't be really accurate anymore and that it probably shouldn't be exported (I tried to use `Size(DAY_SIZE_SQUARE, height)` only to find out that it's not working...).
One option I've been thinking about is to move those size related fields/methods directly to the `Size` class, something like this:
```kotlin
data class Size(@Px val width: Int, @Px val height: Int) {
	companion object {
		internal const val AUTO = Int.MIN_VALUE
		val SQUARE = Size(AUTO, AUTO)
		fun autoWidth(@Px height: Int) = Size(AUTO, height)
	}
}
```
The downside is that this would create some deprecated fields... We could also move just that new method to `Size` class and leave the rest intact, but then it will be inconsistent.
If the method should stay in `CalendarView` class, I cannot think of a better name than the one proposed.

**Comment by kizitonwose** [comment]
> Another thing is that the name DAY_SIZE_SQUARE won't be really accurate anymore

`DAY_SIZE_SQUARE` is deprecated anyways and will be removed soon so no worries about that.

I like the idea of moving it to the Size class. Guess we can do that in the next major release but have it this way for now.

**Review by kizitonwose — APPROVED** [review]

**Pull request merged**